Source/WebCore/style/StyleAdjuster.cpp

```diff
@@ -775,9 +775,13 @@ static bool hasEffectiveDisplayNoneForDisplayContents(const Element& element)
 {
     using namespace ElementNames;
 
+    // According to the CSS Display spec[1], nested <svg> elements, <g>,
+    // <use>, and <tspan> elements are not rendered and their children are
+    // "hoisted". For other elements display:contents behaves as display:none.
     // https://drafts.csswg.org/css-display-3/#unbox-svg
-    // FIXME: <g>, <use> and <tspan> have special (?) behavior for display:contents in the current draft spec.
-    if (is<SVGElement>(element))
+    if (!is<SVGElement>(element))
+        return false;
+    if (RefPtr svgElement = dynamicDowncast<SVGElement>(element); (svgElement && svgElement->isOutermostSVGSVGElement()) || (!is<SVGSVGElement>(element) && !svgElement->hasTagName(SVGNames::tspanTag) && !svgElement->hasTagName(SVGNames::gTag) && !svgElement->hasTagName(SVGNames::useTag)))
```

Comment:
it would be interesting to discover why the tests results currently failing have changed. 

* accessibility/slider-with-lost-renderer.html
* dom/html/document-renderobject-null-crash.html
* editing/selection/redo-selection-restore-different-document-crash.html
* [imported/w3c/web-platform-tests/css/css-content/content-none-input.html](https://wpt.fyi/css/css-content/content-none-input.html) ([live](https://wpt.live/css/css-content/content-none-input.html))
* [imported/w3c/web-platform-tests/css/css-display/accessibility/display-contents-role-and-label.html](https://wpt.fyi/css/css-display/accessibility/display-contents-role-and-label.html) ([live](https://wpt.live/css/css-display/accessibility/display-contents-role-and-label.html))
* [imported/w3c/web-platform-tests/css/css-display/display-contents-suppression-dynamic-001.html](https://wpt.fyi/css/css-display/display-contents-suppression-dynamic-001.html) ([live](https://wpt.live/css/css-display/display-contents-suppression-dynamic-001.html))
* [imported/w3c/web-platform-tests/css/css-display/display-contents-svg-anchor-child.html](https://wpt.fyi/css/css-display/display-contents-svg-anchor-child.html) ([live](https://wpt.live/css/css-display/display-contents-svg-anchor-child.html))
* [imported/w3c/web-platform-tests/css/css-display/display-contents-unusual-html-elements-none.html](https://wpt.fyi/css/css-display/display-contents-unusual-html-elements-none.html) ([live](https://wpt.live/css/css-display/display-contents-unusual-html-elements-none.html))
* [imported/w3c/web-platform-tests/html/rendering/widgets/button-layout/computed-style.html](https://wpt.fyi/html/rendering/widgets/button-layout/computed-style.html) ([live](https://wpt.live/html/rendering/widgets/button-layout/computed-style.html))
* [imported/w3c/web-platform-tests/mathml/relations/css-styling/display-contents.html](https://wpt.fyi/mathml/relations/css-styling/display-contents.html) ([live](https://wpt.live/mathml/relations/css-styling/display-contents.html))


Specifically for the WPT tests. Are the tests wrong and need to be fix. 
Or is the change not giving what is expected from the spec.

Reply:
For example, in the test https://wpt.live/mathml/relations/css-styling/display-contents.html
this is about mathML and there is nothing which seems related to SVG. flaky or something is affecting the change?

Reply:
Another strange one.
https://wpt.fyi/html/rendering/widgets/button-layout/computed-style.html

nothing seems to relate to SVG in the test. 

```
93 PASS computed display of <input type=reset> with display: contents
94 PASS computed display of <input type=button> with display: contents
95 PASS computed display of <input type=submit> with display: contents
96 PASS computed display of <input type=color> with display: contents
97 PASS computed display of <input type=image> with display: contents
 93 FAIL computed display of <input type=reset> with display: contents assert_equals: expected "none" but got "contents"
 94 FAIL computed display of <input type=button> with display: contents assert_equals: expected "none" but got "contents"
 95 FAIL computed display of <input type=submit> with display: contents assert_equals: expected "none" but got "contents"
 96 FAIL computed display of <input type=color> with display: contents assert_equals: expected "none" but got "contents"
 97 FAIL computed display of <input type=image> with display: contents assert_equals: expected "none" but got "contents"
```

Reply:
Relevant tests too.
http://wpt.live/css/css-display/display-contents-svg-elements.html

Reply:
<img width="1722" alt="Screenshot 2024-12-23 at 12 38 11" src="https://github.com/user-attachments/assets/2ae6503a-0a34-4688-aa0d-c878468d64d4" />

Reply:
also interesting
https://github.com/WebKit/WebKit/blob/5592179d5613f47a4b4659f02bf9713c54086670/Source/WebCore/style/StyleAdjuster.cpp#L834-L879